number_theory/numeric.h

```diff
@@ -43,6 +45,54 @@ std::pair<T, T> exgcd(T a, T b) {
   return std::make_pair(xa, ya);
 }
 
+// Accumulates values according to the binary representation of |binary|.
+// For each bit of |binary|, from lower to higher, it calls |update| to
+// accumulate the values. The initial value is |init|.
+template <class T, class U>
+U binary_accumulate(T binary,
```

Comment:
It seems that this functions acts like `reduce` function in functional programming. The difference is that `reduce` takes two consecutive elements from a list and apply `function` to them, and add the result back. In this case, you are doing the same thing bitwise. So, maybe we call this a `bitwise_reduce`, `binary_reduce` function?

According to [wikipedia](https://en.wikipedia.org/wiki/Fold_(higher-order_function)), this functionality is named as `fold (also termed reduce, accumulate, aggregate, compress, or inject)`. We can see that `reduce`, `accumulate`, and `aggregate` are also used interchangeably for it. Maybe, your naming is fine.

Reply:
I think this function is like a variant of std::accumulate for binary values, so I want to keep this name.